lib/propolis/src/hw/testdev.rs

```diff
@@ -0,0 +1,122 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+//! Devices intended for testing purposes.
+//!
+//! These devices do things which are generally unwanted in real life, such as
+//! "intentionally breaking Propolis", "intentionally breaking the guest OS", or
+//! some combination of the two.
+
+use crate::inventory::Entity;
+use crate::migrate::*;
+
+use serde::{Deserialize, Serialize};
+
+use slog::info;
+use std::sync::{
+    atomic::{AtomicUsize, Ordering},
+    Arc,
+};
```

Comment:
style turbo-nit: elsewhere in the repo, imports have been grouped as:
1. std
2. crate-internal
3. external